mvrx/src/main/kotlin/com/airbnb/mvrx/DeliveryMode.kt

```diff
@@ -34,3 +40,15 @@ object RedeliverOnStart : DeliveryMode()
  * have the same id.
  */
 class UniqueOnly(val subscriptionId: String) : DeliveryMode()
+
+/**
+ * The subscription will receive the most recent state update when transitioning from locked to unlocked states (stopped -> started) after the
+ * caller's provided custom behavior via [flowConfiguration]. If [distinctOnly] is true, this will behave like []UniqueOnly].
+ *
+ * Likewise, when a [MavericksView] resubscribes after a configuration change the most recent update will be emitting after processing the desired
+ * custom behavior.
+ *
+ * @param subscriptionId A uniqueIdentifier for this subscription. It is an error for two unique only subscriptions to
+ * have the same id.
+ */
+class Custom(val subscriptionId: String, val distinctOnly: Boolean, val flowConfiguration: (Flow<*>, DeliveryMode) -> Flow<*>) : DeliveryMode()
```

Comment:
I think it's a little weird to expose both `flowConfiguration` and `distinctOnly`, since if we are providing custom flow configuration then they can apply `distinctUntilChanged` easily enough on their own, and our goal is not to parameterize each possible flow operator

Reply:
hmm i think if we do that we would need flowConfiguration to also handle `flowWhenStarted`. do we actually want to do that?